pkg/chains/signing.go

```diff
@@ -204,6 +217,9 @@ func (o *ObjectSigner) Sign(ctx context.Context, tektonObj objects.TektonObject)
 				}
 				if err := b.StorePayload(ctx, tektonObj, rawPayload, string(signature), storageOpts); err != nil {
 					logger.Error(err)
+					if (signableType.Type() == "TaskRunArtifact" || signableType.Type() == "PipelineRunArtifact") && o.Recorder != nil {
+						o.Recorder.RecordErrorMetric(ctx, "storage")
+					}
```

Comment:
The `signableType.Type() == "..."` logic appears a lot. Could this be abstracted? E.g.:
```go
if err := b.StorePayload(ctx, tektonObj, rawPayload, string(signature), storageOpts); err != nil {
    logger.Error(err)
    o.recordErrorMetrics(ctx, signableType.Type(), "storage")
}

// ...

func (o *ObjectSigner) recordErrorMetric(ctx context.Context, signableType string, errType string) {
    recordableTypes := []string{"TaskRunArtifact",  "PipelineRunArtifact"}
    if o.Recorder != nil && slices.Contains(recordableTypes, signableType) {
        o.Recorder.RecordErrorMetric(ctx, errType)
    }
}
```

Reply:
i have abstracted it:
`func (o *ObjectSigner) shouldRecordError(kind string) bool {
	return kind == "TaskRunArtifact" || kind == "PipelineRunArtifact"
}`

Reply:
Let me know if it looks good. TY!

Reply:
Thanks @mathur07! It's a bit better, however there is still a fair bit of duplication. Any objection to consolidating the `shouldRecordErrorMetric` and `o.Recorder != nil` into the same function like in my example?

Reply:
+1
Use constant for TaskRunArtifact and PipelineRunArtifact kinds